spec/classes/backend_spec.rb

```diff
@@ -247,6 +252,20 @@
            is_expected.to compile.and_raise_error(/Do not define both license_source and license_content/)
         end
       end
+
+      context 'checks defined' do
```

Comment:
we should have similar tests for each type.

Reply:
Because of amount of tests needed and fact resources actually defined in a different class, I moved tests for all 19 parameters to separate file.